api/app/controllers/mno_enterprise/jpi/v1/admin/invoices_controller.rb

```diff
@@ -54,6 +64,45 @@ def last_commission_amount
 
     private
 
+    def invoice_adjustments
+      bills = @invoice.bills
+      params[:invoice][:adjustments].each do |adjustment|
+        attributes = adjustment.permit(*ADJUSTMENT_ATTRIBUTES).merge(bill_attributes)
+        # create a new bill only if the param adjustment has no :id
+        MnoEnterprise::Bill.create(attributes) unless adjustment.key?(:id)
+        # update the bill if fields have changed
+        update_bill(adjustment) if adjustment.key?(:id) && bills.any? { |b| b.id == adjustment[:id] && (b.billable != adjustment[:billable_description] || b.price_cents != adjustment[:price_cents])}
```

Comment:
Also, following @alachaum comments on https://github.com/maestrano/maestrano-hub/pull/672 you will have to use relationships instead of id. As it's done for Subscription for example
https://github.com/maestrano/mno-enterprise/blob/4.0/api/lib/mno_enterprise/concerns/controllers/jpi/v1/subscriptions_controller.rb

```
    subscription = MnoEnterprise::Subscription.new(subscription_update_params)
    subscription.relationships.organization = MnoEnterprise::Organization.new(id: parent_organization.id)
    subscription.relationships.user = MnoEnterprise::User.new(id: current_user.id)
    subscription.relationships.product_pricing = MnoEnterprise::ProductPricing.new(id: params[:subscription][:product_pricing_id])
    subscription.relationships.product_contract = MnoEnterprise::ProductContract.new(id: params[:subscription][:product_contract_id])
    subscription.save
```